src/App.tsx

```diff
@@ -1,16 +1,24 @@
 import './App.scss';
+import { useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [movies, setMovies] = useState(moviesFromServer);
+
+  const onAdd = (movie: Movie) => {
+    (setMovies([...movies, movie]));
```

Comment:
If the new state is computed using the previous state, you can pass a function to setMovies
```suggestion
    (setMovies((movies) => [...movies, movie]));
```